sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserFreeUpRequestValidator.java

```diff
@@ -0,0 +1,87 @@
+package org.sunbird.common.request;
+
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.responsecode.ResponseCode;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class UserFreeUpRequestValidator extends BaseRequestValidator {
+
+    private Request request;
+    private static List<String> identifiers = new ArrayList<>();
+    static {
+        identifiers.add("email");
+        identifiers.add("phone");
+    }
+
+    private static final int ERROR_CODE = ResponseCode.CLIENT_ERROR.getResponseCode();
+
+
+    public static UserFreeUpRequestValidator getInstance(Request request) {
+        return new UserFreeUpRequestValidator(request);
+    }
+
+    private UserFreeUpRequestValidator(Request request) {
+        this.request = request;
+    }
+
+    public void validate() {
+        validateIdPresence();
+        validateIdentifier();
+    }
+
+
+    private void validateIdPresence() {
+        validateParam(
+                (String) request.getRequest().get(JsonKey.ID),
+                ResponseCode.mandatoryParamsMissing,
+                JsonKey.ID);
+    }
+
+    private void validateIdentifier() {
+        validatePresence();
+        validateObject();
+        validateSubset();
+    }
+
+
+    private void validatePresence() {
+        if (!request.getRequest().containsKey(JsonKey.IDENTIFIER)) {
+            throw new ProjectCommonException(
+                    ResponseCode.mandatoryParamsMissing.getErrorCode(),
+                    MessageFormat.format(ResponseCode.mandatoryParamsMissing.getErrorMessage(), JsonKey.IDENTIFIER),
+                    ResponseCode.CLIENT_ERROR.getResponseCode());
+        }
+
+    }
+
+    private void validateObject() {
+        Object identifierType = request.getRequest().get(JsonKey.IDENTIFIER);
+        if (!(identifierType instanceof List)) {
```

Comment:
what will happen if identifierType is null

Reply:
```{
    "request": {
        "id": "061bc053-f744-487e-a060-ae91456bf2b9",
        "identifier":null
    }
}```
Response will be 
```{
    "id": "",
    "ver": "private",
    "ts": "2019-07-26 14:00:25:480+0530",
    "params": {
        "resmsgid": null,
        "msgid": "d219fca5-cd8d-4457-a679-1d3bee090acb",
        "err": "DATA_TYPE_ERROR",
        "status": "DATA_TYPE_ERROR",
        "errmsg": "Data type of identifier should be List."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}```